pkg/capture/capture_manager.go

```diff
@@ -515,6 +519,10 @@ func (cm *Manager) rotate(ctx context.Context, writeoutChan chan<- capturetypes.
 			// Lock the running capture in order to safely perform rotation tasks
 			if err := mc.capLock.Lock(); err != nil {
 				logger.Errorf("failed to establish rotation three-point lock: %s", err)
+				if err := mc.close(); err != nil {
```

Comment:
This was missing (I think because I was worried that we might end up with duplicate calls to `close()`. However, this is now reasonably safe due to the additional capture Mutex.